File: Questions/Songs.java
import java.util.*;

public class Songs {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] songsInput = scanner.nextLine().split(" ");
        String[] animationsInput = scanner.nextLine().split(" ");
        Map<String, Integer> hashsongs = new HashMap<>();
        for (String song : songsInput) {
            int index = song.indexOf(':');
            String key = song.substring(0, index);
            int value = Integer.parseInt(song.substring(index + 1));
            hashsongs.put(key, value);
        }
        Map<Integer, String> hashanimations = new TreeMap<>(); // TreeMap to maintain sorted order by default
        for (String animation : animationsInput) {
            int index = animation.indexOf(':');
            String key = animation.substring(0, index);
            int value = Integer.parseInt(animation.substring(index + 1));
            hashanimations.put(value, key);
        }
        
        List<String> res = new ArrayList<>();

        for (Map.Entry<String, Integer> songEntry : hashsongs.entrySet()) {
            String songKey = songEntry.getKey();
            int songValue = songEntry.getValue();
            
            for (Map.Entry<Integer, String> animationEntry : hashanimations.entrySet()) {
                int animationValue = animationEntry.getKey();
                String animationKey = animationEntry.getValue();
                
                if (songValue % animationValue == 0) {
                    String result = animationKey + ":" + (songValue / animationValue);
                    res.add(result);
                    break;
                }
            }
        }
        System.out.println(res);
    }
}

File: Questions/Python/songs.py
songs = list(map(str, input().split()))
animations = list(map(str, input().split()))

hashsongs = {}
hashanimations = {}

for i in songs:
    index = i.index(':')
    hashsongs[i[:index]] = int(i[index+1:])

for i in animations:
    index = i.index(':')
    hashanimations[int(i[index+1:])] = i[:index]


res = []

for i in hashsongs.keys():
    val = hashsongs[i]
    for j in sorted(hashanimations.keys()):
        if val%j == 0:
            s = hashanimations[j] + ":" + str(val//j)
            res.append(s)
            break

print(res)


